utils/patterns/pattern.go

```diff
@@ -0,0 +1,698 @@
+package patterns
```

Comment:
Where is the OpenAPI schema for this resource?  Let's ensure that this package, structs, functions are based directly on that schema.

Reply:
Below is the spec for `Meshery Patterns`,  the structs defined here are internal (except DryRun).
https://github.com/meshery/meshery/blob/db401c30f56ff47eae6849bf19aad0c32d8dac2a/server/helpers/swagger.yaml#L2036